### ISSUE TITLE
SQLSTATE[42000]: Syntax error or access violation

### ISSUE THREAD

**Issue by cheve** [micwallace/wallacepos#40]
To acces admin account display this message:

SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'wallacepos.i.name' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

It allows changes to the settings but does not add Locations & Devices.

Run new version wallacepos in:
Lubuntu 16.04 LTS
Kernel Linux 4.4.0-21-generic (i686)
Apache/2.4.18 (Ubuntu)
MySQL 5.7.12-0ubuntu1
PHP: 7.0.4-7ubuntu2

Excuse me work that I am generating ;-)
Thanks.

**Comment by cheve** [comment]
If you’ve recently upgraded your MySQL server version to MySQL 5.7.5+ and you notice that some of your queries that contain GROUP BY clause, suddenly stop working, then it could be because of a change that was made in MySQL 5.7.5+.
[FAQ sql modes](https://dev.mysql.com/doc/refman/5.7/en/faqs-sql-modes.html)

Remove mode "only_full_group_by" from sql_mode. To acces admin account display this message:
stime must be numeric
etime must be numeric

It allows changes to the settings but does not add Locations & Devices.

I go good path??? Thanks...

**Comment by cheve** [comment]
modifycate and new message of error, modifycate and new message of error
I think we will have to revert to earlier versions of SQL.
I return to ubuntu 14.04 LTS

**Comment by ais-one** [comment]
thanks for indicating, I may try out on xampp with mysql 5.7.7 onwards to see the effect...

**Comment by cheve** [comment]
Wallacepos working without problems in Ubuntu 14.04.4 LTS and Ubuntu 15.10. Early to Ubuntu 16.04

Bufff ais-one!!!
Dependencias, clausulas, columnas, violaciones... I lost.
Thanks.

**Comment by micwallace** [comment]
I'll likely be moving infrastructure to 16.04 in the near future and will look at the issue.

**Comment by micwallace** [comment]
Can confirm that changes in mysql 5.7 caused this issue.

**Comment by ais-one** [comment]
I can download 16.04 on virtual box and try it out later.

**Comment by cheve** [comment]
Yes, it seems that changes to MySQL 5.7 cause this problem. Other applications also have problems after switching to SQL 5.7. ChromisPos for example.
Wallacepos working without problems in Ubuntu 14.04.4 LTS and Ubuntu 15.10.

Ubuntu 14.04 LTS
Apache 2.4.7
MYSQL 5.5.49
PHP 5.5.9

Ubuntu 15.10
Apache 2.4.12
MYSQL 5.6.30
PHP 5.6.11

Ubuntu 16.04 LTS
Apache 2.4.18
MySQL 5.7.12
PHP: 7.0.4

Thanks

**Comment by micwallace** [comment]
eae6ec4 should fix this issue, although I have not yet tested on mysql 5.7

**Comment by cheve** [comment]
I hope Orange pi II Plus to install ubuntu 16 LTS. We will test with modifications.

Thank you.

**Comment by cheve** [comment]
Hi. It is not stable ubuntu 16 in orange pi + 2. A try debian 8.5

**Comment by micwallace** [comment]
As a temp fix you can remove mysql_full_group_by from sql mode:

stackoverflow.com/questions/23921117/disable-only-full-group-by

**Comment by epiqueras** [comment]
I'm also getting the same error after successfully updating to version 1.4.1.

**Comment by micwallace** [comment]
Yes you still need to disable only_full_group_by in mysql.

**Comment by darrenarauz21** [comment]
Hacer:

$ sudo nano /etc/mysql/my.cnf
Copiar y pegar:

[mysqld]
sql_mode=STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
Al final del archivo

$ sudo service mysql restart

Aún así saldrán nuevos problemas. Así que lo conveniente es instalar una versión más antigua de Ubuntu y mysql.
